src/index.ts

```diff
@@ -1,73 +1,70 @@
 import * as core from '@actions/core';
-import * as github from '@actions/github';
-import { getStaticBundleSizes, getDynamicBundleSizes, getMarkdownTable } from './bundle-size';
+import { context, getOctokit } from '@actions/github';
+import { getStaticBundleSizes, getMarkdownTable } from './bundle-size';
 
-import { createOrReplaceComment } from './comments';
-import { createOrReplaceIssue } from './issue';
+import { createOrUpdateCommentPartially } from './comments';
+import { createOrUpdateIssuePartially } from './issue';
 import { downloadArtifactAsJson } from './download-artifacts';
 import { uploadJsonAsArtifact } from './upload-artifacts';
+import { createPartialBundleInfo } from './create-partial-bundle-info';
+import { determineAppName } from './determine-app-name';
 
-const ARTIFACT_NAME = 'next-bundle-analyzer';
+const ARTIFACT_NAME_PREFIX = 'next-bundle-analyzer__';
 const FILE_NAME = 'bundle-sizes.json';
-const DYNAMIC_FILE_NAME = 'dynamic-bundle-sizes.json';
+const COMMENT_TITLE = '## Bundle Sizes';
 
 async function run() {
   try {
-    const workflowId = core.getInput('workflow-id', { required: true });
-    const baseBranch = core.getInput('base-branch') || 'master';
     const workingDir = core.getInput('working-directory') || '';
+    const appName = determineAppName(workingDir);
+    const artifactName = `${ARTIFACT_NAME_PREFIX}${appName}`;
 
-    const octokit = github.getOctokit(process.env.GITHUB_TOKEN || '');
-    const issueNumber = github.context.payload.pull_request?.number;
+    const octokit = getOctokit(process.env.GITHUB_TOKEN || '');
 
-    console.log(`> Downloading bundle sizes from ${baseBranch}`);
-    const masterBundleSizes = (await downloadArtifactAsJson(
+    const {
+      data: { default_branch },
+    } = await octokit.rest.repos.get({ ...context.repo });
+
+    const issueNumber = context.payload.pull_request?.number;
+
+    console.log(`> Downloading bundle sizes from ${default_branch}`);
+    const referenceBundleSizes = (await downloadArtifactAsJson(
       octokit,
-      baseBranch,
-      workflowId,
-      ARTIFACT_NAME,
+      default_branch,
+      artifactName,
       FILE_NAME,
     )) || { sha: 'none', data: [] };
-    console.log(masterBundleSizes);
-    const masterDynamicBundleSizes = (await downloadArtifactAsJson(
-      octokit,
-      baseBranch,
-      workflowId,
-      ARTIFACT_NAME,
-      DYNAMIC_FILE_NAME,
-    )) || { sha: 'none', data: [] };
-    console.log(masterDynamicBundleSizes);
+    console.log(referenceBundleSizes);
 
     console.log('> Calculating local bundle sizes');
     const bundleSizes = getStaticBundleSizes(workingDir);
     console.log(bundleSizes);
-    const dynamicBundleSizes = getDynamicBundleSizes(workingDir);
-    console.log(dynamicBundleSizes);
```

Comment:
Why have these been removed?

Reply:
Two main reasons I suppose.
1. It's a source of noise that doesn't really help with evaluating whether the PR increases the page size. This is mostly due to when you update dependencies, the new files use a different hash in the file names. See https://github.com/transferwise/launchpad-frontend/pull/933 for an example.
2. Current implementation won't work with the latest version of artifacts, since I believe it uploads the same artifact multiple times. This is addressable but I'm not sure it's worth it. https://github.blog/changelog/2023-12-14-github-actions-artifacts-v4-is-now-generally-available/